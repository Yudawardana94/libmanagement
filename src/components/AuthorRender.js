import {StyleSheet, Text, View} from 'react-native';
import React from 'react';

const AuthorRender = ({authorData}) => {
  return (
    <View style={styles.mr4} key={authorData.name}>
      <Text>Author: </Text>
      <View style={styles.authorWrapper}>
        {authorData.map((author, idx) => (
          <Text style={styles.fs12} key={author.name}>
            {author.name}
            {idx === authorData.length - 1 ? '' : ', '}
          </Text>
        ))}
      </View>
    </View>
  );
};

export default AuthorRender;

const styles = StyleSheet.create({
  authorWrapper: {
    flexDirection: 'row',
    flex: 1,
    flexWrap: 'wrap',
  },
  fs12: {fontSize: 12},
  mr4: {marginTop: 4},
});
