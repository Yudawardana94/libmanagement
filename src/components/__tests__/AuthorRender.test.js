import React from 'react';
import {create, act} from 'react-test-renderer';
import AuthorRender from '../AuthorRender';

const authorMockData = [
  {
    url: 'el.authorsurl',
    name: 'el.authors.name',
  },
];
const onHandleNavigation = () => {
  return true;
};
const tree = create(
  <AuthorRender
    authorData={authorMockData}
    onHandleNavigation={onHandleNavigation}
  />,
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
