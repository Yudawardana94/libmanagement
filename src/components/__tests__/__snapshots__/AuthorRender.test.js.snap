// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 4,
    }
  }
>
  <Text>
    Author: 
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      el.authors.name
    </Text>
  </View>
</View>
`;
